### ISSUE TITLE
Rule Proposal: Disallow style in Single File components

### ISSUE THREAD

**Issue by leipert** [vuejs/eslint-plugin-vue#927]
**Please describe what the rule should do:**
Some projects might have different CSS pipelines, and don't want to utilize `<style>` in single file components. A rule like e.g. `no-style` one could forbid it for their project.

**What category should the rule belong to?**

- [ ] Enforces code style
- [x] Warns about a potential error
- [ ] Suggests an alternate way of doing something
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**
```vue
<style>
/* Example */
</style>
```

**Additional context**
